File: eVetCare.Services/Database/Pet.cs
using System;
using System.Collections.Generic;

namespace eVetCare.Services.Database;

public partial class Pet
{
    public int PetId { get; set; }

    public int OwnerId { get; set; }

    public string Name { get; set; } = null!;

    public int SpeciesId { get; set; }

    public string? Breed { get; set; }

    public int? Age { get; set; }

    public double? Weight { get; set; }

    public string? PhotoUrl { get; set; }

    public int? GenderId { get; set; }

    public bool IsActive { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual Gender? Gender { get; set; }

    public virtual ICollection<MedicalRecord> MedicalRecords { get; set; } = new List<MedicalRecord>();

    public virtual User Owner { get; set; } = null!;

    public virtual ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();

    public virtual Species Species { get; set; } = null!;
}

File: UI/evetcare_mobile/lib/models/pet.dart
class Pet {
  final int ownerId;
  final String ownerFirstName;
  final String ownerLastName;
  final String ownerEmail;
  final String ownerPhoneNumber;
  final String name;
  final int speciesId;
  final String breed;
  final int genderId;
  final int age;
  final double weight;
  final String? photo;

  Pet({
    required this.ownerId,
    required this.ownerFirstName,
    required this.ownerLastName,
    required this.ownerEmail,
    required this.ownerPhoneNumber,
    required this.name,
    required this.speciesId,
    required this.breed,
    required this.genderId,
    required this.age,
    required this.weight,
    this.photo,
  });

  Map<String, dynamic> toJson() {
    return {
      'ownerId': ownerId,
      'ownerFirstName': ownerFirstName,
      'ownerLastName': ownerLastName,
      'ownerEmail': ownerEmail,
      'ownerPhoneNumber': ownerPhoneNumber,
      'name': name,
      'speciesId': speciesId,
      'breed': breed,
      'genderId': genderId,
      'age': age,
      'weight': weight,
      if (photo != null) 'photo': photo,
    };
  }
}
